backend/tracim_backend/lib/webdav/resources.py

```diff
@@ -68,143 +70,224 @@ def wrapper(self: "_DAVResource", *arg, **kwarg) -> typing.Callable:
     return decorator
 
 
-class RootResource(DAVCollection):
+def get_workspace_resource(
+    path: str,
+    environ: typing.Dict,
+    workspace: Workspace,
+    label: str,
+    tracim_context: "WebdavTracimContext",
+) -> "WorkspaceResource":
+    return WorkspaceResource(
+        path=path, environ=environ, workspace=workspace, tracim_context=tracim_context, label=label,
+    )
+
+
+def get_content_resource(
+    content: Content,
+    path: str,
+    environ: typing.Dict,
+    workspace: Workspace,
+    tracim_context: "WebdavTracimContext",
+) -> _DAVResource:
     """
-    RootResource ressource that represents tracim's home, which contains all workspaces
+    Helper to get correct content webdav resource according to content type
     """
+    if content.type == content_type_list.Folder.slug:
+        return FolderResource(
+            path=path,
+            environ=environ,
+            workspace=workspace,
+            content=content,
+            tracim_context=tracim_context,
+        )
+    elif content.type == content_type_list.File.slug:
+        return FileResource(
+            path=path, environ=environ, content=content, tracim_context=tracim_context,
+        )
+    else:
+        return OtherFileResource(
+            path=path, environ=environ, content=content, tracim_context=tracim_context
+        )
 
-    def __init__(self, path: str, environ: dict, tracim_context: "WebdavTracimContext"):
-        super(RootResource, self).__init__(path, environ)
+
+class WebdavContainer(ABC):
+    """
+    A Webdav Container should implement theses methods.
+    """
+
+    @abstractmethod
+    def createEmptyResource(self, file_name: str):
+        pass
+
+    @abstractmethod
+    def createCollection(self, label: str) -> "FolderResource":
+        pass
+
+    @abstractmethod
+    def getMemberNames(self) -> [str]:
+        pass
+
+    @abstractmethod
+    def getMember(self, label: str) -> _DAVResource:
+        pass
+
+    @abstractmethod
+    def getMemberList(self) -> [_DAVResource]:
+        pass
```

Comment:
Are you free to name these methods as you wish?

If so:
 - "member" is confusing with respect to what is usually a member in Tracim 
 - maybe use snake_case?

Reply:
No, it's related to wsgidav : https://github.com/mar10/wsgidav . Version 2.* that we do use use CamelCase, we do need to update the dependencies to support snake_case. I don't do it now, as this may be an heavy change.
see this ticket https://github.com/tracim/tracim/issues/2024